### ISSUE TITLE
convert non builtin models in the marketplace to new API (Trac #1046)

### ISSUE THREAD

**Issue by butlerpd** [SasView/sasmodel-marketplace#14]
!SasView 4.2 changes completely how orientation is done and the API to the sasmodels interface thus changes so that all old models will fail.  This is a problem for plugins, including anything downloaded from the marketplace.  The solution is twofold: 

* First, in ticket #11, sasmodels needs to recognize old models and ignore the 2D (maintaining backward compatibility is way too hard and has a number of other issues that make it not worth doing).  This should allow all old plugins that were used for 1D to continue to work.

* The second part is this ticket and is to convert the few custom models currently uploaded to the marketplace that have real 2D code to use the new API.  Also, as per ticket #13, we should change the name slightly so as to keep both versions on the marketplace with a note in the documentation for now indicating the version difference.

Migrated from http://trac.sasview.org/ticket/1046
```json
{
    "status": "closed",
    "changetime": "2018-10-29T13:33:09",
    "_ts": "2018-10-29 13:33:09.301321+00:00",
    "description": "!SasView 4.2 changes completely how orientation is done and the API to the sasmodels interface thus changes so that all old models will fail.  This is a problem for plugins, including anything downloaded from the marketplace.  The solution is twofold: \n\n* First, in ticket #1043, sasmodels needs to recognize old models and ignore the 2D (maintaining backward compatibility is way too hard and has a number of other issues that make it not worth doing).  This should allow all old plugins that were used for 1D to continue to work.\n\n* The second part is this ticket and is to convert the few custom models currently uploaded to the marketplace that have real 2D code to use the new API.  Also, as per ticket #1045, we should change the name slightly so as to keep both versions on the marketplace with a note in the documentation for now indicating the version difference.",
    "reporter": "butler",
    "cc": "",
    "resolution": "wontfix",
    "workpackage": "SasModels Redesign",
    "time": "2017-12-12T18:58:48",
    "component": "sasmodels Markeplace",
    "summary": "convert non builtin models in the marketplace to new API",
    "priority": "critical",
    "keywords": "",
    "milestone": "SasView 4.2.0",
    "owner": "butler",
    "type": "task"
}
```

**Comment by butlerpd** [comment]
Trac update at `2017/12/12 19:06:16`: 
* **butler** changed component from "SasView" to "sasmodels Markeplace"
* **butler** changed workpackage from "SasView Bug Fixing" to "SasModels Redesign"

**Comment by butlerpd** [comment]
Trac update at `2018/02/13 14:43:33`: 
* **butler** changed owner from "" to "butler"
* **butler** changed status from "new" to "assigned"

**Comment by butlerpd** [comment]
Trac update at `2018/04/02 23:17:16`: **butler** changed priority from "critical" to "major"

**Comment by butlerpd** [comment]
Trac update at `2018/09/27 14:40:46`: 
* **butler** commented:

At this point all remaining tickets should either be blockers or critical.  Critical would be those that do not stop the actual release but need to be dealt with soon after such as updating the marketplace. Tickets such as this are blockers in that if not fixed the docs need to have an appropriate warning in them for this release.  Once that is fixed the ticket moves to 4.3 if it is not closed.



* **butler** changed priority from "major" to "critical"

**Comment by butlerpd** [comment]
Trac update at `2018/10/29 13:33:09`: 
* **butler** commented:

After some discussion, it is clear that the remaining portion of this ticket is ill advised.  Changing user uploaded models out from under them, while could be helpful to making their models stay up to date, would also preclude the ability to use this as a repository of the actual model they used in a paper.  The zenodo doi does not include their custom code (but does include the built in model code).  This just reinforces the need to refactor the marketplace to provide the !SasView versions a model is compatible with.

As agreed on the last fortnightly call this ticket is therefore being closed as won't fix


* **butler** changed resolution from "" to "wontfix"
* **butler** changed status from "assigned" to "closed"
